app/Actions/Album/SetSorting.php

```diff
@@ -2,14 +2,20 @@
 
 namespace App\Actions\Album;
 
+use App\Exceptions\ModelDBException;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
 class SetSorting extends Action
 {
-	public function do(string $albumID, ?string $sortingCol, ?string $sortingOrder): bool
+	/**
+	 * @throws ModelDBException
+	 * @throws ModelNotFoundException
+	 */
+	public function do(string $albumID, ?string $sortingCol, ?string $sortingOrder): void
```

Comment:
why is this method taking a `$albumID` instead of a BaseAlbum instead ?
(just to be consistent with the other Action calls)

Reply:
Because the action calls are not used by the controllers anymore. At least not for the trivial setters. For this specific case see

https://github.com/LycheeOrg/Lychee/pull/1217/files#diff-64646ce0183575e7ffc0f7292df2b7611839781641acddd7792a4c029e5404c1R298-R302

Actually, most of the actions can be nuked. I didn't do that, because I did not know if you still want to keep them for the Livewire part. But currently they are dead code.

Reply:
But if I don't use the actions in the Livewire side, wouldn't that lead to code duplication ? :thinking:

Reply:
No, it would not. Actually most of the actions are completely ridiculous. They create around 20 lines of code of indirection instead of one. Moreover, they are not type safe at all. For example.

Setting the title of an album now is basically:

```php
class AlbumController {
  public function setTitle(Album $album, string $title): void {
    $album->title = $title;
  }
}
```

I left out some parts, but I think the general idea is clear. The method is trivial and concise. The actual "magic" is a single line. The "Action" classes basically did the same, but in an unnecessary complicated manner.

```php
class AlbumSetter {
  private string $attribute;
  
  protected function __construct(string $attribute) {
    $this->attribute = $attribute;
  }

  /**
   * @param Album $album  the album whose attribute shall be set
   * @param mixed $value  the new value of the attribute
   */
  public function set(Album $album, $value): void {
    $album->{$this->attribute} = $value;
  }
}

class AlbumTitleAction extends AlbumSetter {
  public function __construct() {
    parent::__construct('title');
  }
}

class AlbumController {
  public function setTitle(Album $album, string $title, AlbumTitleAction $action): void {
    $action->set($album, $title);
  }
}
```

The latter approach is completely insane. The `AlbumSetter` class does nothing more than setting a single attribute of the object. Moreover, `$album->{$this->attribute} = $value` is dangerous in two ways:

 1. As we use runtime reflection here, we don't know if `$this->attribute` actually denotes a proper attribute of `$album`. We have no IDE completion and any static code analyzer will simply skip this part, because it cannot do anything.
 2. `$value` is of mixed type, because we don't know whether the requested attribute is a string, an integer, a boolean, a float or another object (e.g. a DateTime object).

And worst of all: it does not save anything. Instead of directly calling `$album->title = $title` in the controller method we call `$action->set($album, $title);`.

Reply:
Resolved by simple deletion of offending class.